lib/shared/src/configuration/auth-resolver.ts

```diff
@@ -0,0 +1,101 @@
+import type {
+    AuthCredentials,
+    ClientConfiguration,
+    ExternalAuthCommand,
+    ExternalAuthProvider,
+} from '../configuration'
+import type { ClientSecrets } from './resolver'
+
+export function normalizeServerEndpointURL(url: string): string {
+    return url.endsWith('/') ? url : `${url}/`
+}
+
+async function executeCommand(cmd: ExternalAuthCommand): Promise<string> {
+    if (typeof process === 'undefined' || !process.version) {
+        throw new Error('Command execution is only supported in Node.js environments')
+    }
+
+    const { exec } = await import('node:child_process')
+    const { promisify } = await import('node:util')
+    const execAsync = promisify(exec)
+
+    const command = cmd.commandLine.join(' ')
+
+    // No need to check error code, promisify causes exec to throw in case of errors
+    const { stdout } = await execAsync(command, {
+        shell: cmd.shell,
+        timeout: cmd.timeout,
+        windowsHide: cmd.windowsHide,
+        env: cmd.environment ? { ...process.env, ...cmd.environment } : process.env,
```

Comment:
You can splice from undefined, null, etc. without problems (at least in JavaScript... if TypeScript complains I guess you could `(cmd.environment || {})`.)

```suggestion
        env: { ...process.env, ...cmd.environment },
```